react/src/components/SourceCodeViewer.tsx

```diff
@@ -0,0 +1,97 @@
+import { useThemeMode } from '../hooks/useThemeMode';
+import Flex from './Flex';
+import { CheckOutlined, CopyOutlined } from '@ant-design/icons';
+import { Button, theme } from 'antd';
+import { Highlight, themes } from 'prism-react-renderer';
+import { useRef, useState } from 'react';
+// @ts-ignore
+import { CopyToClipboard } from 'react-copy-to-clipboard';
+
+interface SourceCodeViewerProps {
+  // In case of using this component in a web component, the props should be parsed from the web component.
+  codeBlock: string;
+  language: string;
+  fontSize?: string;
+  minHeight?: string;
+}
+
+const SourceCodeViewer: React.FC<SourceCodeViewerProps> = ({
+  codeBlock,
+  language,
+  fontSize,
+  minHeight,
+}) => {
+  const timer = useRef<number>();
+  const { token } = theme.useToken();
+  const { isDarkMode } = useThemeMode();
+  const [isCopied, setIsCopied] = useState(false);
+  const [isMouseEntered, setIsMouseEntered] = useState(false);
+  const HighLightedCode = (
+    <Highlight
+      theme={isDarkMode ? themes.jettwaveDark : themes.jettwaveLight}
+      code={codeBlock || ''}
+      language={language || ''}
+    >
+      {({ className, style, tokens, getLineProps, getTokenProps }) => (
+        <pre
+          style={{
+            ...style,
+            width: '100%',
+            minHeight: minHeight,
+            fontSize: `${fontSize}`,
+            padding: token.padding,
+            borderRadius: token.borderRadius,
+            whiteSpace: 'pre-wrap',
+            wordBreak: 'break-word',
+          }}
+        >
+          {tokens.map((line, i) => (
+            <Flex key={i} {...getLineProps({ line })}>
+              {line.map((token, key) => (
+                <span key={key} {...getTokenProps({ token })} />
+              ))}
+            </Flex>
+          ))}
+        </pre>
+      )}
+    </Highlight>
+  );
+
+  return (
+    <Flex
+      onMouseEnter={() => setIsMouseEntered(true)}
+      onMouseLeave={() => setIsMouseEntered(false)}
+    >
+      <CopyToClipboard
+        text={codeBlock}
+        onCopy={() => {
+          setIsCopied(true);
+          clearTimeout(timer.current);
+          timer.current = window.setTimeout(() => {
+            setIsCopied(false);
+          }, 2000);
+        }}
+      >
+        <Button
+          style={{
+            position: 'absolute',
+            bottom: 24,
+            right: 10,
+            cursor: 'pointer',
+            zIndex: 1000,
```

Comment:
Please remove this if it is not needed. If it is needed, please use `token.zIndexBase` to set the zIndex, for example, `token.zIndexBase + 1`. Setting a zIndex too high, even if not necessary, can break the UI related to the z-index stack (modals, popup, ...).